gitimerge.py

```diff
@@ -3714,20 +3714,25 @@ def cmd_diagram(parser, options):
         merge_frontier.write_html(html, merge_state.name)
         html.close()
     sys.stdout.write(
-        'Key:\n'
+        'Key:                                                    Map:\n'
+        )
+    sys.stdout.write(
+        '  * = merge done manually                                 o-0-1-2..-%(len1)d  %(tip1)s\n'
+        '  . = merge done automatically                              1\n'
+        '  # = conflict that is currently blocking progress          ..\n'
+        '  @ = merge was blocked but has been resolved               %(len2)d\n'
```

Comment:
I suggest right-justifying the `..` and the `13` on the vertical axis, to make the enclosed space look a bit more rectangular. I think the following delta will do it:

```suggestion
        '  # = conflict that is currently blocking progress         ..\n'
        '  @ = merge was blocked but has been resolved           %(len2)4d\n'
```